ui_framework/components/button/button.test.js

```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import {
+  shallow,
+  render,
+} from 'enzyme';
+import sinon from 'sinon';
+
+import {
+  KuiButton,
+} from './button';
+
+describe('KuiButton', () => {
+  describe('Baseline', () => {
+    test('is rendered', () => {
+      const $button = shallow(
+        <KuiButton />
+      );
+
+      expect($button)
+        .toMatchSnapshot();
+    });
+  });
+
+  describe('Props', () => {
+    describe('type', () => {
+      describe('basic', () => {
+        test('renders the basic class', () => {
+          const $button = render(
+            <KuiButton type="basic" />
+          );
+
+          expect($button)
+            .toMatchSnapshot();
+        });
+      });
+
+      describe('hollow', () => {
+        test('renders the hollow class', () => {
+          const $button = render(
+            <KuiButton type="hollow" />
+          );
+
+          expect($button)
+            .toMatchSnapshot();
+        });
+      });
+
+      describe('danger', () => {
+        test('renders the danger class', () => {
+          const $button = render(
+            <KuiButton type="danger" />
+          );
+
+          expect($button)
+            .toMatchSnapshot();
+        });
+      });
+
+      describe('primary', () => {
+        test('renders the primary class', () => {
+          const $button = render(
+            <KuiButton type="primary" />
+          );
+
+          expect($button)
+            .toMatchSnapshot();
+        });
+      });
+    });
```

Comment:
I find this to be really verbose, but I think that comes down to a stylistic difference.  This is one more reason why enums for types would work well here.  You could loop through the enum and make sure every option was rendered, so you couldn't accidentally add a string in that propType but forget to map it to a class name inside the button class.

something like this:

```
describe('all button types', () => {
  test('are rendered', () => {
    foreach (type in KuiButton.TYPES) {
      const $button = render(<KuiButton type="{ type }" />);
      expect($button).toMatchSnapshot();
    }
  });
});
```

Then you never have to touch the tests for every new type added or removed, it'll all just work, and still all be tested.

barring the enum style, you could still do something like:
```
describe('all button types', () => {
  test('are rendered', () => {
    foreach (type in ['danger', 'basic', 'primary]) {
      const $button = render(<KuiButton type="{ type }" />);
      expect($button).toMatchSnapshot();
    }
  });
});
```

But then still you have to remember to manually update the tests for every type added or removed.

One more thing... if you did add a string but forget to map it to a class name, I don't think that will throw an error here?  Because it'll match the snapshot?  unless we added a throw new Error('unhandled type') in the code? Or maybe jest tests will catch the invalid propType warning and treat that as a test failure?